### PR TITLE
[CAZ-1738] [CAZ-1739] Add direct debit

### PR DESCRIPTION
**Pull request by miqs1992**
## Motivation and Context

https://eaflood.atlassian.net/browse/CAZ-1738
https://eaflood.atlassian.net/browse/CAZ-1739

## Checklist:

- [ ] It contains only changes required by issue (does not contain other PR)
- [ ] Includes link to a issue (if apply)
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: app/models/mandate.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#
+class Mandate
+  # Initializer method. Assigns :data
+  #
+  # ==== Params
+  #
+  # * +zoneId+ - uuid, CleanAirZone ID
+  # * +zoneName+ - string, name of the CAZ
+  # * +mandateId+ - uuid, mandate ID
+  # * +status+ = string, status of the mandate eg. 'pending'
```
Comment (likeuwill) [inline app/models/mandate.rb]: `+status+ -`

File: app/views/debits/new.html.haml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+= link_to 'Back', @mandates.size.zero? ? dashboard_path : debits_path, class: 'govuk-back-link'
+
+%main.govuk-main-wrapper#main-content{role: 'main'}
+  .govuk-grid-row
+    .govuk-grid-column-two-thirds
+      - if alert
+        = render 'common/error_summary',
+          error_msg_div: '#caz-selection-error',
+          error_msg: alert
```
Comment (likeuwill) [inline app/views/debits/new.html.haml]: Align parameters with first

File: app/views/fleets/index.html.haml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-= link_to 'Back to dashboard', dashboard_path, class: 'govuk-back-link'
+= back_to_dashboard
```
Comment (likeuwill) [inline app/views/fleets/index.html.haml]: `back_link_to_dashboard` would be better naming

File: spec/requests/debits/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'DebitsController - #index', type: :request do
```
Comment (likeuwill) [inline spec/requests/debits/index_spec.rb]: `RSpec.describe 'DebitsController - GET #index' do`
same for all tests, no need to use `type` and add `method`

File: app/views/debits/new.html.haml (hunk 2 above)
Comment (miqs1992) [reply app/views/debits/new.html.haml]: I don't understand this. Do you mean: 
```
= render 'common/error_summary',
               error_msg_div: '#caz-selection-error',
               error_msg: alert
```

**Pull request merged**